src/pkg/archive/archive_test.go

```diff
@@ -5,48 +5,197 @@
+	entries, err := os.ReadDir(dstDir)
+	require.NoError(t, err, "ReadDir failed")
+	if len(entries) != 0 {
+		t.Errorf("expected no files extracted; found %d entries", len(entries))
+	}
+}
+
+func TestDecompress_UnarchiveAll(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	tmp := t.TempDir()
+
+	// Create a nested tar archive inside another directory
+	innerDir := filepath.Join(tmp, "inner")
+	require.NoError(t, os.Mkdir(innerDir, 0o755))
+	innerFile := filepath.Join(innerDir, "foo.txt")
+	writeFile(t, innerFile, "nested content")
+
+	innerTar := filepath.Join(tmp, "inner.tar")
+	require.NoError(t, Compress(ctx, []string{innerFile}, innerTar, CompressOpts{}))
+
+	// Now create an archive containing the inner tar
+	outerDir := filepath.Join(tmp, "outer")
+	require.NoError(t, os.Mkdir(outerDir, 0o755))
+	outerTar := filepath.Join(tmp, "outer.tar")
+	require.NoError(t, os.Rename(innerTar, filepath.Join(outerDir, "inner.tar")))
+	require.NoError(t, Compress(ctx, []string{filepath.Join(outerDir, "inner.tar")}, outerTar, CompressOpts{}))
+
+	// Decompress with UnarchiveAll
+	outDir := filepath.Join(tmp, "out")
+	opts := DecompressOpts{UnarchiveAll: true}
+	require.NoError(t, Decompress(ctx, outerTar, outDir, opts))
+
+	// Should have extracted foo.txt from the nested tar
+	found := false
+	err := filepath.Walk(outDir, func(path string, _ os.FileInfo, _ error) error {
+		if filepath.Base(path) == "foo.txt" {
+			found = true
+			content := readFile(t, path)
+			if content != "nested content" {
+				t.Errorf("expected 'nested content', got %q", content)
+			}
+		}
+		return nil
+	})
+	require.NoError(t, err, "Walk failed")
+	if !found {
+		t.Error("foo.txt not found after UnarchiveAll")
+	}
+}
+
+func TestDecompress_OverwriteExisting(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	tmp := t.TempDir()
+
+	// Create a file and archive it
+	origFile := filepath.Join(tmp, "orig.txt")
+	writeFile(t, origFile, "original")
+	archivePath := filepath.Join(tmp, "archive.tar.gz")
+	require.NoError(t, Compress(ctx, []string{origFile}, archivePath, CompressOpts{}))
+
+	// Decompress once
+	outDir := filepath.Join(tmp, "out")
+	require.NoError(t, Decompress(ctx, archivePath, outDir, DecompressOpts{}))
+	outFile := filepath.Join(outDir, "orig.txt")
+	if got := readFile(t, outFile); got != "original" {
+		t.Fatalf("expected original, got %q", got)
+	}
+
+	// Overwrite the file with new content and re-archive
+	writeFile(t, origFile, "new content")
+	archivePath2 := filepath.Join(tmp, "archive2.tar.gz")
+	require.NoError(t, Compress(ctx, []string{origFile}, archivePath2, CompressOpts{}))
+
+	// Decompress with OverwriteExisting
+	opts := DecompressOpts{OverwriteExisting: true}
+	require.NoError(t, Decompress(ctx, archivePath2, outDir, opts))
+	if got := readFile(t, outFile); got != "new content" {
+		t.Errorf("expected overwritten file to have 'new content', got %q", got)
+	}
+}
```

Comment:
In an ideal world these tests would be expressed using the table driven tests pattern: https://go.dev/wiki/TableDrivenTests#parallel-testing. This PR does fix a CVE though, so I don't see this as a blocking request, just a quality of life improvement for maintainers down the line if we can budget the time. Thoughts?